src/devices/Nrf24l01/Nrf24l01.cs

```diff
@@ -0,0 +1,767 @@
+        /// </summary>
+        /// <param name="payload">Size, from 0 to 32</param>
+        internal void SetRxPayload(byte payload)
+        {
+            if (payload > 32 || payload < 0)
+            {
+                throw new ArgumentOutOfRangeException(nameof(payload), $"{nameof(payload)} needs to be in the range of 0 to 32.");
+            }
+
+            _gpio.Write(_ce, PinValue.Low);
+
+            Span<byte> writeData = stackalloc byte[1] { payload };
+
+            Write(Command.NRF_W_REGISTER, Register.NRF_RX_PW_P0, writeData);
+            Write(Command.NRF_W_REGISTER, Register.NRF_RX_PW_P1, writeData);
+            Write(Command.NRF_W_REGISTER, Register.NRF_RX_PW_P2, writeData);
+            Write(Command.NRF_W_REGISTER, Register.NRF_RX_PW_P3, writeData);
+            Write(Command.NRF_W_REGISTER, Register.NRF_RX_PW_P4, writeData);
+            Write(Command.NRF_W_REGISTER, Register.NRF_RX_PW_P5, writeData);
+
+            _gpio.Write(_ce, PinValue.High);
+        }
+
+        /// <summary>
+        /// Set nRF24L01 Receive Packet Size
+        /// </summary>
+        /// <param name="pipe">Pipe, form 0 to 5</param>
+        /// <param name="payload">Size, from 0 to 32</param>
+        internal void SetRxPayload(byte pipe, byte payload)
+        {
+            if (payload > 32 || payload < 0)
+            {
+                throw new ArgumentOutOfRangeException(nameof(payload), $"{nameof(payload)} needs to be in the range of 0 to 32.");
+            }
+
+            if (pipe > 5 || pipe < 0)
+            {
+                throw new ArgumentOutOfRangeException(nameof(pipe), $"{nameof(pipe)} needs to be in the range of 0 to 5.");
+            }
+
+            Span<byte> writeData = stackalloc byte[] { (byte)((byte)Command.NRF_W_REGISTER + (byte)Register.NRF_RX_PW_P0 + pipe), payload };
+
+            _gpio.Write(_ce, PinValue.Low);
+
+            Write(writeData);
+
+            _gpio.Write(_ce, PinValue.High);
+        }
+
+        /// <summary>
+        /// Read nRF24L01 Receive Packet Size
+        /// </summary>
+        /// <param name="pipe">Pipe, form 0 to 5</param>
+        /// <returns>Size</returns>
+        internal byte ReadRxPayload(byte pipe)
+        {
+            if (pipe > 5 || pipe < 0)
+            {
+                throw new ArgumentOutOfRangeException(nameof(pipe), $"{nameof(pipe)} needs to be in the range of 0 to 5.");
+            }
+
+            Span<byte> readData = WriteRead(Command.NRF_R_REGISTER, (Register)((byte)Register.NRF_RX_PW_P0 + pipe), 1);
+
+            return readData[0];
+        }
+
+        /// <summary>
+        /// Set nRF24L01 Auto Acknowledgment (All Pipe)
+        /// </summary>
+        /// <param name="isAutoAck">Is Enable</param>
+        internal void SetAutoAck(bool isAutoAck)
+        {
+            _gpio.Write(_ce, PinValue.Low);
+
+            if (isAutoAck)
+            {
+                Write(Command.NRF_W_REGISTER, Register.NRF_EN_AA, 0x3F);
```

Comment:
please describe bits in this register and if makes sense please use binary literal (`0b0000_0000`)